src/psyclone/dynamo0p3.py

```diff
@@ -307,33 +314,41 @@ def qr_basis_alloc_args(first_dim, basis_fn):
     '''
     if basis_fn["shape"] not in VALID_QUADRATURE_SHAPES:
         raise InternalError(
-            "Unrecognised shape ({0}) specified in "
+            "Unrecognised shape ('{0}') specified in "
             "dynamo0p3.qr_basis_alloc_args(). Should be one of: "
             "{1}".format(basis_fn["shape"], VALID_QUADRATURE_SHAPES))
 
+    if basis_fn["qr_var"]:
+        qr_var = "_" + basis_fn["qr_var"]
+    else:
+        # We have no name for the QR variable (as will be the case when
+        # generating a kernel stub) so invent one.
+        qr_var = "_qr_" + basis_fn["shape"].split("_")[-1]
```

Comment:
This line does not appear to be covered by tests.

Reply:
Oh yes - I altered the way I set-up the qr-var variable name so that it now always has a value, even if we're dealing with a kernel stub.